packages/engine/Source/Scene/Model/CustomShader.js

```diff
@@ -209,6 +209,17 @@ function CustomShader(options) {
    */
   this.uniformMap = buildUniformMap(this);
 
+  /**
+   * The map of uniform names to an object that contains booleans which will influence how these uniforms are handled
+   * in the rendering pipeline. E.g. if a uniform has the boolean `shouldConvertToModelCoordinates` set to `true`, its
+   * value will be converted to model coordinates before being passed to the shader (only applicable for `Cartesian3`s).
+   *
+   * @type {Object<string, Object<string, boolean>>}
```

Comment:
I think `Object` is not generic. Although this is JS, you should be able to use TS syntax and types in JSDoc comments. So, it should be possible to use `Record`.
Furthermore, is the value always of type `Record<string, boolean>` or wouldn't it be possible to have other values than `boolean`? I.e., would `Record<string, unknown>` be more fitting?

Reply:
Hmm, for now I would keep it as `Boolean` because I think booleans make the most sense and any other additions to the object would need code changes anyways. I think it would be better to adapt this in the future and then use something like `boolean|string` instead of `unknown`.
(There's also quite the probability that we will never add anything else to this object anyways.. 😂)

Regarding the typing:
I think it makes the most sense to keep it in the way that Cesium does it, so we don't introduce contradicting/different ways of typing. I fear we'll end up with a less-maintainable mess otherwise. 🙈 
That's why I simply copied the typing from the other properties in this class (which are "generic" Objects 🤷).